lib/trino/client/statement_client.rb

```diff
@@ -208,7 +208,8 @@ def faraday_get_with_retry(uri, &block)
             return response
           end
 
-          if response.status != 503  # retry only if 503 Service Unavailable
+          # retry if 502, 503, 504 according to the trino protocol
+          if response.status == 502 || response.status == 503 || response.status == 504
```

Comment:
```suggestion
          unless [502, 503, 504].include?(response.status)
```

Reply:
Hi, @exoego Thanks for the review. I have applied your review comment code.